aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file aesd-circular-buffer.c
+ * @brief Functions and data related to a circular buffer imlementation
+ *
+ * @author Dan Walkes
+ * @date 2020-03-01
+ * @copyright Copyright (c) 2020
+ *
+ */
+
+#ifdef __KERNEL__
+#include <linux/string.h>
+#else
+#include <string.h>
+#endif
+
+#include "aesd-circular-buffer.h"
+
+#define MAX_NUMBER 9
```

Comment:
@aneesh1298 how does this relate to AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED?  It would be better to define based on this since these are dependent on each other.